### ISSUE TITLE
Hide controls and use implicit mouse controls instead

### ISSUE THREAD

**Issue by protist** [antroids/application-title-bar#3]
In the Active Window Control plasmoid from Plasma 5, there was an option to hide the "Elements", i.e. the buttons for close/minimise/maximise. We could instead control the window using other actions, e.g. middle-click to close, scroll down to minimise, scroll up to maximise. This would be brilliant to be added to this plasmoid too.

**Comment by r4** [comment]
I think this is more or less asking for the same feature I was about to post about. It would be nice to have the buttons only show up when a window is maximized. Otherwise, they are hidden.

**Comment by antroids** [comment]
Mouse actions added to v0.3.0

**Comment by Okona** [comment]
> I think this is more or less asking for the same feature I was about to post about. It would be nice to have the buttons only show up when a window is maximized. Otherwise, they are hidden.

I second that! The now defunct Window Buttons widget offered exactly that behavior. This behaviour is especially useful combined with borderless maximized windows

**Comment by antroids** [comment]
> > I think this is more or less asking for the same feature I was about to post about. It would be nice to have the buttons only show up when a window is maximized. Otherwise, they are hidden.
> 
> I second that! The now defunct Window Buttons widget offered exactly that behavior. This behaviour is especially useful combined with borderless maximized windows

This is already implemented.
You can update or reinstall the widget if the options are still unavailable.
There are a lot of issues with widgets updates on new Plasma.

After release 0.3.0 you can remove all buttons by dragging them to the bin at Appearance config and assign actions to the mouse events at Behaviour config page.

![Screenshot_Settings_2](https://github.com/antroids/application-title-bar/assets/36043354/0982afc0-437f-40c6-8408-716c1ac4db62)

**Comment by Okona** [comment]
I was referring to the comment from user r4 here. 
It was about hiding the Widget elements except when the current application is fullscreen.

It might not be totally the correct issue here. Should I create a new issue?

Btw, I am Using 0.3.5 which I freshly installed.

**Comment by antroids** [comment]
Oh, sorry. I didn't notice that this is different request with the first post. I'll add this option too.

**Comment by antroids** [comment]
> I was referring to the comment from user r4 here. It was about hiding the Widget elements except when the current application is fullscreen.
> 
> It might not be totally the correct issue here. Should I create a new issue?
> 
> Btw, I am Using 0.3.5 which I freshly installed.

Added "Show for only maximized" option in 0.3.7

**Comment by antroids** [comment]
> What about hiding the controls unless the window is maximized, but keep the title no matter what?

Do you have a use case for that?
Should the title be for last active window or maximized?

**Comment by aydencook03** [comment]
> > What about hiding the controls unless the window is maximized, but keep the title no matter what?
> 
> Do you have a use case for that? Should the title be for last active window or maximized?

Any active window already has control buttons, but what I'd like to do is hide the window border when the window is maximized. This would mean that the widget's controls are only needed when the window is maximized.

**Comment by antroids** [comment]
> > > What about hiding the controls unless the window is maximized, but keep the title no matter what?
> > 
> > 
> > Do you have a use case for that? Should the title be for last active window or maximized?
> 
> Any active window already has control buttons, but what I'd like to do is hide the window border when the window is maximized. This would mean that the widget's controls are only needed when the window is maximized.

Yes, and any active window has window title. Why you would like to see the window title in the panel? I'm asking because not clear what title should be displayed: from active window or from maximized.

**Comment by protist** [comment]
> Any active window already has control buttons, but what I'd like to do is hide the window border when the window is maximized.

FWIW I never have window decorations, even on un-maximised windows. Hence, I never have buttons at any time.

Also, for me the lack of buttons doesn't necessarily relate to this issue. Even without having buttons, it's still possible for users to interact with the widget using middle-click etc, rather than clicking on visible buttons.
